spring-context/src/main/java/org/springframework/context/support/AbstractApplicationContext.java

```diff
@@ -262,9 +262,8 @@ public String getApplicationName() {
 	 * Typically done during initialization of concrete context implementations.
 	 * <p>Default is the object id of the context instance.
 	 */
-	public void setDisplayName(String displayName) {
-		Assert.hasLength(displayName, "Display name must not be empty");
-		this.displayName = displayName;
+	public void setDisplayName(@Nullable String displayName) {
```

Comment:
`getDisplayName()` isn't actually nullable, so I guess we can leave this setter non-null as well?